### ISSUE TITLE
Problem using mssql, parameterized query, and leading quotes

### ISSUE THREAD

**Issue by leedm777** [ADOdb/ADOdb#420]
I've got the following code:

```php
function find_by_name($name) {
  $sql = "SELECT * FROM some_table WHERE name = ?";
  return $db->execute($sql, array($name));
}
```

It works most of the time (`find_by_name("dave")`), but fails when the parameter begins with a quotation mark (`find_by_name("'wichcraft")`).

I think the problem is on [adodb-mssql.inc.php#L817](https://github.com/ADOdb/ADOdb/blob/master/drivers/adodb-mssql.inc.php#L817). It avoids running `qstr()` if the param string starts with a `'`.

I'm kinda stuck on an older version of ADOdb. Are there any options here for a workaround?

**Comment by mnewnham** [comment]
The mssql driver is still in active support, as it is used for the linux/sql server connection, so a fix for this is still relevant.

Why not change the expression to 
````
if (  substr($v,0,1) != "'" OR substr($v,-1,1) != ""'") THEN quote string
````

**Comment by leedm777** [comment]
@mnewnham I don't know enough about why it's even doing what it's doing to know if that's an appropriate fix or not. I can certainly propose a PR that corrects the behavior for my use case, but without knowing more about why it's doing what it's doing, it's just as likely to break for someone else.

**Comment by mnewnham** [comment]
I have committed a fix to both 5.20 and master
